# Data Analyst Associate Practical Exam Submission

**You can use any tool that you want to do your analysis and create visualizations. Use this template to write up your summary for submission.**

You can use any markdown formatting you wish. If you are not familiar with Markdown, read the [Markdown Guide](https://s3.amazonaws.com/talent-assets.datacamp.com/Markdown+Guide.pdf) before you start.



### Jack Sechler - Data Analyst Associate Practical Exam

## TASK 1 - Data Validation & Cleaning



**Original Dataset**

In [295]:
# Importing Required Libraries and overview of data prior to cleaning
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.style as sty
import plotly.express as px
import seaborn as sns
df = pd.read_csv('food_claims_2212.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   claim_id              2000 non-null   int64  
 1   time_to_close         2000 non-null   int64  
 2   claim_amount          2000 non-null   object 
 3   amount_paid           1964 non-null   float64
 4   location              2000 non-null   object 
 5   individuals_on_claim  2000 non-null   int64  
 6   linked_cases          1974 non-null   object 
 7   cause                 2000 non-null   object 
dtypes: float64(1), int64(3), object(4)
memory usage: 125.1+ KB


In [296]:
#Checking for columns with missing values
df.isna().sum()

claim_id                 0
time_to_close            0
claim_amount             0
amount_paid             36
location                 0
individuals_on_claim     0
linked_cases            26
cause                    0
dtype: int64

In [297]:
# Understanding what this table looks like
df.head()

claim_id  time_to_close  ... linked_cases      cause
0         1            317  ...        False    unknown
1         2            195  ...         True    unknown
2         3            183  ...         True       meat
3         4            186  ...        False       meat
4         5            138  ...        False  vegetable

[5 rows x 8 columns]

In [298]:
#Checking Table Values before Cleaning
print(len(df['claim_id'].unique()))
print(df['linked_cases'].unique())
print(df['location'].unique())
print(df['cause'].unique())

2000
[False True nan]
['RECIFE' 'FORTALEZA' 'SAO LUIS' 'NATAL']
['unknown' 'meat' 'vegetable' ' Meat' 'VEGETABLES']


**Data Validation & Cleaning for each column**

In [299]:
# Validating Time to Close column
print(f"Checking if there are negative values in time_to_close column: {(df.time_to_close < 0).sum()}")

# Validating claim_amount column
# This column indicates the Brazilian national currency. First, we need to replace this with a blankspace, then convert this string to float type. 
df['claim_amount'] = df['claim_amount'].apply(lambda x: x.replace('R$ ', ''))
df['claim_amount'] = df['claim_amount'].astype(float)
print(f'\nChecking that all values in claim_amount column are positive: {(df.claim_amount > 0).all()}')

# Validating amount_paid column
# There were 36 missing values in the 'Amount_Paid' column so following the instructions I replaced those with the overall median amount paid
median_amount_paid = df['amount_paid'].median()
df['amount_paid'] = df['amount_paid'].fillna(value=median_amount_paid)
print(f'\nChecking that all values in amount_paid are positive: {(df.amount_paid >= 0).all()}')

# Validating Location column
print(f'\nCategories of the location column: {df.location.unique()}')

# Validating Individuals_on_Claim column
print(f'\nChecking that all values in the individuals_on_claim column are positive: {(df.individuals_on_claim >= 0).all()}')

# Validating Linked_Cases column
# There are 26 missing values in the 'linked_cases' column, so here we will need to replace these with "FALSE"
df['linked_cases'] = df['linked_cases'].fillna(value=False)
print(f"\nChecking status options for the linked_cases column: {df['linked_cases'].unique()}")

# Validating/Cleaning Causes Column
# There are some edits that needed to be made. There were some causes listed as MEAT or VEGATABLES and that didn't align with the three given options of meat, vegetables, and unknown. So adjusting there.

df.loc[df.cause == 'VEGETABLES', 'cause'] = 'vegetable'
df.loc[df.cause == ' Meat', 'cause'] = 'meat'
print(f'\nChecking categories in the causes column after amending: {df.cause.unique()}')

Checking if there are negative values in time_to_close column: 0

Checking that all values in claim_amount column are positive: True

Checking that all values in amount_paid are positive: True

Categories of the location column: ['RECIFE' 'FORTALEZA' 'SAO LUIS' 'NATAL']

Checking that all values in the individuals_on_claim column are positive: True

Checking status options for the linked_cases column: [False  True]

Checking categories in the causes column after amending: ['unknown' 'meat' 'vegetable']


In [300]:
#Checking how the table looks after cleaning
df.head()

claim_id  time_to_close  ...  linked_cases      cause
0         1            317  ...         False    unknown
1         2            195  ...          True    unknown
2         3            183  ...          True       meat
3         4            186  ...         False       meat
4         5            138  ...         False  vegetable

[5 rows x 8 columns]

In [301]:
# Verifying that columns have been updated
# Verifying no missing values in any column
df.info()
df.isna().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   claim_id              2000 non-null   int64  
 1   time_to_close         2000 non-null   int64  
 2   claim_amount          2000 non-null   float64
 3   amount_paid           2000 non-null   float64
 4   location              2000 non-null   object 
 5   individuals_on_claim  2000 non-null   int64  
 6   linked_cases          2000 non-null   bool   
 7   cause                 2000 non-null   object 
dtypes: bool(1), float64(2), int64(3), object(2)
memory usage: 111.5+ KB


claim_id                0
time_to_close           0
claim_amount            0
amount_paid             0
location                0
individuals_on_claim    0
linked_cases            0
cause                   0
dtype: int64

A. This dataset contains 2000 rows and 8 columns. Analysis of the values & columns below.

1. There are 2000 unique **Claim ID's** and matches the description given.
2. All values in **Time_to_Close** column are positive and there were no missing values.
3. The **claim_amount** column was in object data type, so as a string, I changed it to a float data type.
4. In the **amount_paid** column, there were 36 missing values, which I replaced with the overall median amount paid.
5. There weren't any missing values in the **location** column, and the four values matched those given in the prompt. 
6. All values in the **individuals_on_claim** column had a minimum of 1 person on the claim and there weren't any missing values.
7. In the **linked_cases** column, there were 26 missing values, which I replaced with "FALSE" in alignment with the instructions.
8. The **causes** column contained some entries labeled "VEGETABLES" and "Meat", when the listed causes were "vegetable",  "meat", or "unknown". So I replaced "VEGETABLES" and "Meat" with the original, prescribed values "vegetable" and "meat" respectively. There weren't any missing values here but if there were, they'd be replaced with "unknown."

B.  There were only 2 columns with missing values. This was 'amount_paid' which had 36 missing values, and 'linked_cases' which had 26 missing values.

C. How I updated the columns to match given description
-To prepare the claim_amount column for analysis, I removed the string "R$" at the beginning of the values in the claim_amount column and then changed the type to float, converting it to numeric.
-In the amount_paid column, I replaced all missing values with the overall median amount paid.
-In the linked_cases column, I replaced all missing values with "FALSE".
-In the cause column, there were some entries labeled "VEGETABLES" and "Meat", when the listed causes were "vegetable",  "meat", or "unknown". So I replaced "VEGETABLES" and "Meat" with "vegetable" and "meat" respectively.
-The remaining columns ('claim_id', 'time_to_close', 'location', and 'individuals_on_claim') matched the description and didn't need any updating or replacing.


## TASK 2 - Visualization Showing Number of Claims in Each Location

![Screenshot 2023-03-27 at 5.20.39 PM](Screenshot%202023-03-27%20at%205.20.39%20PM.png)


A. The category of the variable location with the highest amount of claims was 'RECIFE'.

B. The observations are not balanced across categories of the variable location as 'RECIFE' appears to have nearly three times as many as 'NATAL'. To optimize business and improve efficiency, this organization should try to look into why RECIFE has so many claims.

## TASK 3 - Describe the Distribution of Time to Close for all claims and include Visualization

In [302]:
#Box Plot illustrating the breakdown of time to close and summary statistics

fig = px.box(df, y="time_to_close")
print(f'mean: {df.time_to_close.mean()}')
print(f'median: {df.time_to_close.median()}')
print(f'min: {df.time_to_close.min()}')
print(f'max: {df.time_to_close.max()}')
fig.show()

mean: 185.568
median: 179.0
min: 76
max: 518


In [303]:
#Historgram illustrating the Distribution of Time to Close for All Claims

fig = px.histogram(df, y="time_to_close", nbins=20, marginal="box",
                  title="Distribution of Time to Close across Claims",
                  width=750, labels ={'time_to_close': 'Time to Close'})
print(f'mean: {df.time_to_close.mean()}')
print(f'median: {df.time_to_close.median()}')
print(f'min: {df.time_to_close.min()}')
print(f'max: {df.time_to_close.max()}')
fig.show()

mean: 185.568
median: 179.0
min: 76
max: 518


The distribution of time to close for all claims can be seen through the histogram and box plot above. The distribution of the time to close across all claims is not normal. We can see in the bins of the chart that many of the claims fell between 100-300 days mark. Specifically, the 150-199 days bin had the highest number of occurrences with about 1,051 claims closing in that range. From this, we can see that most claims are resolved between 100 and 300 days, with some clear outliers coming in on the longer end of the spectrum. I've also included some summary statistics, which can also can be seen by hovering over the box plot on the right panel.

## Task 4 - Describe the relationship between time to close and location and include Visualization

In [304]:
fig = px.box(df, x="location", y="time_to_close", 
             labels={
                 "location": "Location",
                 "time_to_close": "Time to Close"
             },
             title="Box Plot of Time to Close across Location")
fig.show()

In [305]:
#Average time to close at each location 
avgtime = df.groupby("location")["time_to_close"].mean()
avgtime

location
FORTALEZA    185.305466
NATAL        185.926829
RECIFE       184.607910
SAO LUIS     187.170213
Name: time_to_close, dtype: float64

In [306]:
#Breakdown of average time to close at each location by cause
df.groupby(["location","cause"])["time_to_close"].mean()

location   cause    
FORTALEZA  meat         182.532544
           unknown      197.094737
           vegetable    171.446809
NATAL      meat         184.211268
           unknown      197.823529
           vegetable    163.372093
RECIFE     meat         180.793765
           unknown      199.459627
           vegetable    162.746575
SAO LUIS   meat         183.458515
           unknown      201.577320
           vegetable    166.478723
Name: time_to_close, dtype: float64

In [307]:
#breakdown of median time to close at each location by cause
df.groupby(["location", "cause"])["time_to_close"].median()

location   cause    
FORTALEZA  meat         179.0
           unknown      188.0
           vegetable    172.0
NATAL      meat         178.5
           unknown      184.5
           vegetable    169.0
RECIFE     meat         176.0
           unknown      186.0
           vegetable    170.0
SAO LUIS   meat         182.0
           unknown      185.0
           vegetable    171.0
Name: time_to_close, dtype: float64

In [308]:
pip install pingouin

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.


In [309]:
import pingouin
alpha = 0.05
pingouin.kruskal(data=df, dv='time_to_close', between='location')

Source  ddof1         H     p-unc
Kruskal  location      3  1.153223  0.764244

As we can see in the charts and graphs above, there is no significant difference between the means of these locations. Because the p-value is 0.76, this is much higher than anything that would indicate a relationship between the two variables (would be closer to 0 if so), so we must accept the null hypothesis. The means of the four locations all fall within a range of four numbers, all being between 184 and 187 days of average time to close. From this analysis of the variables, we deduce that there is no relationship between time to close and location, and any sort of supposed correlation is likely by chance.

## ✅ When you have finished...
-  Publish your Workspace using the option on the left
-  Check the published version of your report:
	-  Can you see everything you want us to grade?
    -  Are all the graphics visible?
-  Review the grading rubric. Have you included everything that will be graded?
-  Head back to the [Certification Dashboard](https://app.datacamp.com/certification) to submit your practical exam